influence_benchmark/backend/hf_backend.py

```diff
@@ -8,48 +8,52 @@
 from influence_benchmark.backend.backend import Backend
 
 
-# TODO: Have a generic backend class that both backend classes inherit from
-class HFBackendMultiton(Backend):
+class HFBackend(Backend):
     """A multiton class for managing multiple instances of the Hugging Face backend. This class is a singleton for each model name.
     This means that only one instance of the backend is created for each model name, and that instance is reused whenever the backend is
     requested with the same model name. This reduces the memory usage of the backend.
     """
 
-    _instances = {}
+    def __init__(self, model_name, device, lora_path=None):
+        self.model = AutoModelForCausalLM.from_pretrained(model_name).half().eval().to(device)
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name, padding_side="left")
+        if lora_path is not None:
+            self.lora = True
 
-    @classmethod
-    def get_instance(cls, model_name, device="cpu"):
-        if model_name not in cls._instances:
-            cls._instances[model_name] = cls._create_instance(model_name, device)
-        return cls._instances[model_name]
+            self.model.load_adapter(lora_path, adapter_name="agent")
+            self.model.disable_adapters()
 
-    @classmethod
-    def _create_instance(cls, model_name, device):
-        instance = super().__new__(cls)
-        instance.model = AutoModelForCausalLM.from_pretrained(model_name).half().eval().to(device)
-        instance.tokenizer = AutoTokenizer.from_pretrained(model_name, padding_side="left")
-        if instance.tokenizer.pad_token is None:
-            instance.tokenizer.pad_token = instance.tokenizer.eos_token
-        instance.device = device
-        return instance
+            self.lora_active = False
+        else:
+            self.lora = False
+        if self.tokenizer.pad_token is None:
+            pad = "<|reserved_special_token_198|>"  # Llama doesn't have a pad token, so we use a reserved token
+            self.pad_id = self.tokenizer.convert_tokens_to_ids(pad)
+            self.tokenizer.pad_token = pad
+            self.tokenizer.pad_token_id = self.pad_id
+            self.model.config.pad_token_id = self.pad_id
+            self.model.generation_config.pad_token_id = self.pad_id
```

Comment:
Seems better to add these as kwargs while creating the model and tokenizer classes, than setting them like this without setters after class creation, otherwise it may cause (hard to detect) problems

One way to do this would be to check at the beginning of `__init__` whether the model is llama, and if so, add the pad_token as an argument to the initialization (assuming that is possible – but it should be?)